### DISCUSSION TITLE
A potentially easier way to get Still Image URL's for struggling users

### DISCUSSION THREAD

**Discussion by potent33** [tsightler/ring-mqtt#534] [Feature Requests]
I have had success with this link.
I found a trick where, by **using a phone or an emulated phone on a browser to use HA, you can retrieve the link from an existing snapshot camera** (**provided that the entity exists** and doesn't register when you try to enter the Still Image URL into RTSP setup, I personally had a issue in the logs stating I had no attributes when trying to use the layout with 'https://myha.mydomain.local/' as the base URL) and use it for the Still Image URL. 

Experiment with the link, sometimes a forward slash and or a full stop after the port is needed, or a full stop after the entity is declared.
The API links given outright by the link on the phone also worked for me 'http://homeassistant.local:8123/api/camera_proxy_stream/camera.front_door_snapshot' which I then modified for my purposes to get the base link but expect some delays in the image for some reason.

This was the link layout I used.

https://homeassistant:8123{{ states.camera.front_door_snapshot.attributes.entity_picture }}

**Comment by potent33** [comment]
(Inspect Element doesn't give an accurate URL for the Snapshot Camera, for some reason this is only possible with an iOS/Android/Emulated Mobile Browser)

**Reply by tsightler** [reply]
I've been considering the possibility of serving the snapshots from the internal express server.  This would theoretically make it easier, at least for addon users.  Although, realistically, it's just the internal URL of your HA instance, with a template where the only variable is the entity name, which is easily found.

Home Assistant does provide an API that I can query the "automatic" internal URL (from Settings...System...Netwokr...Home Assistant URL) but I'm not sure how accurate that is (it's actually wrong in my case for example) and it would only work for addon users.  I could then, at least theoretically, also possibly query the entity name to make a much better "guess".
